### ISSUE TITLE
Issue with OFDMDetectorWithPrior with nulled subcarriers

### ISSUE THREAD

**Issue by rwiesmayr** [NVlabs/sionna#95]
**Describe the bug**
When applying an OFDM resource grid with nulled subcarriers (e.g. DC carriers and guard carriers), any non-zero prior leads to very poor detection performance (e.g. BLER=1). I assume that the problem lies in the indices in `self._data_ind_scatter`, which correspond to indices of data symbols of the whole OFDM resource grid, but not to the indices of the resource grid with removed nulled subcarriers. This becomes problematic in line 504, where the prior template applies `num_effective_subcarriers` in the fourth tensor dimension. My suggestion to fix this issue is to change line 449 to the following:
```
remove_nulled_sc = RemoveNulledSubcarriers(resource_grid)
self._data_ind_scatter = tf.where(remove_nulled_sc(rg_type)==0)
```

**To Reproduce**
1. Apply the MMSE PIC OFDM detector with any ODFM resource grid which includes nulled subcarriers
2. Provide the MMSE PIC with any (e.g. Gaussian) prior
3. Benchmark BER/BLER

**System:**
 - OS: Ubuntu 20.04.4 LTS, CentOS
 - NVIDIA Driver version: 450.203.03, 515.65.01
 - CUDA version: 11.0 and 11.7
 - Python version: 3.8
 - TensorFlow version: 2.7.4

**GPUs:**
 - Model: A100-80GB
 - Memory: 80GB

**Comment by faycalaa** [comment]
Hi Reinhard,

I have done some simulations with `OFDMDetectorWithPrior` in this notebook:
https://gist.github.com/faycalaa/a9623d4d85e2fcea709f7681f29082c5

It seems to work as expected. Could you have a look to help me reproduce this issue?

**Comment by rwiesmayr** [comment]
Hi Faycal,

Thanks for your response! I've just sent you a Jupyter notebook where the issue occurs via email. Hope this helps.

Best regards,
Reinhard

**Comment by gmarcusm** [comment]
fixed in release 0.13.0
